### PR TITLE
Instance segmentation improvements

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Fix empty bbox/mask handling
* Fix validation metrics computation
* Fix validation loss computation
* Make train mAP optional (disabled by default because it is slow)

Will add to docs in a follow-up PR

## How has it been tested?

* Ran ViT-B on COCO

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_transforms/instance_segmentation_transform.py [hunk 1]
```diff
@@ -251,16 +251,25 @@ def __call__(
...
-        # Albumentations doesn't do this automatically, but it removes bboxes and
-        # class labels that correspond to such masks.
+
+        # Albumentations can drop bboxes if they are out of the image after the transform.
+        # It also automatically drops the corresponding class labels and indices but
+        # this doesn't work for masks. So we need to filter them out manually.
         masks = transformed["masks"]
-        masks_tensor = masks if isinstance(masks, Tensor) else torch.stack(masks)
-        masks_tensor = masks_tensor[masks_tensor.sum(dim=(1, 2)) > 0]
+        masks = [masks[i] for i in transformed["indices"]]
```
Comment (stegmuel) [inline src/lightly_train/_transforms/instance_segmentation_transform.py:261]: When albumentations drops a bbox it also drop the corresponding index from `indices`?

Comment (guarin) [reply src/lightly_train/_transforms/instance_segmentation_transform.py:261]: Yes if you set it in bbox params:
```
    bbox_params: BboxParams = BboxParams(
        format="yolo", label_fields=["class_labels", "indices"]
    )
```
